cli/src/pcluster/api/controllers/cluster_logs_controller.py

```diff
@@ -0,0 +1,194 @@
+                            latest log events are returned first. (Defaults to &#39;false&#39;.)
+    :type start_from_head: bool
+    :param limit: The maximum number of log events returned. If you don&#39;t specify a value, the maximum is as many
+                  log events as can fit in a response size of 1 MB, up to 10,000 log events.
+    :type limit:
+    :param start_time: The start of the time range, expressed in ISO8601 format
+                       (e.g. &#39;2021-01-01T20:00:00.000Z&#39;). Events with a timestamp equal to this time or later
+                       than this time are included.
+    :type start_time: str
+    :param end_time: The end of the time range, expressed in ISO8601 format (e.g. &#39;2021-01-01T20:00:00.000Z&#39;).
+                     Events with a timestamp equal to or later than this time are not included.
+    :type end_time: str
+
+    :rtype: GetClusterLogEventsResponseContent
+    """
+    if start_time:
+        _validate_timestamp(start_time, "start_time")
+    if end_time:
+        _validate_timestamp(end_time, "end_time")
+
+    cluster = Cluster(cluster_name)
+    try:
+        if not check_cluster_version(cluster):
+            raise BadRequestException(
+                f"cluster '{cluster_name}' belongs to an incompatible ParallelCluster major version."
+            )
+    except StackNotFoundError:
+        raise NotFoundException(
+            f"cluster '{cluster_name}' does not exist or belongs to an incompatible ParallelCluster major version."
+        )
+    log_events = cluster.get_log_events(
+        log_stream_name,
+        start_time=start_time,
+        end_time=end_time,
+        start_from_head=start_from_head,
+        limit=limit,
+        next_token=next_token,
+    )
+
+    def convert_log_event(event):
+        del event["ingestionTime"]
+        event["timestamp"] = to_iso_time(event["timestamp"])
+        return LogEvent.from_dict(event)
+
+    events = [convert_log_event(e) for e in log_events.events]
+    return GetClusterLogEventsResponseContent(
+        events=events, next_token=log_events.next_ftoken, prev_token=log_events.next_btoken
+    )
+
+
+@configure_aws_region()
+@convert_errors()
+def get_cluster_stack_events(cluster_name, region=None, next_token=None):
+    """
+    Retrieve the events associated with the stack for a given cluster.
+
+    :param cluster_name: Name of the cluster
+    :type cluster_name: str
+    :param region: AWS Region that the operation corresponds to.
+    :type region: str
+    :param next_token: Token to use for paginated requests.
+    :type next_token: str
+
+    :rtype: GetClusterStackEventsResponseContent
+    """
+    cluster = Cluster(cluster_name)
+    stack_events = cluster.get_stack_events(next_token=next_token)
+
+    def convert_event(event):
+        event = {k[0].lower() + k[1:]: v for k, v in event.items()}
```

Comment:
why do we need to lowerize the first character of the key?

Reply:
This is the format that the generated model expects when it's creating the response object (vs what is provided from AWS) so we are just translating between the two.